pkgs/development/tools/unityhub/default.nix

```
@@ -1,45 +1,120 @@
-    url = "https://archive.org/download/unity-hub-${version}/UnityHub.AppImage";
-    sha256 = "07nfyfp9apshqarc6pgshsczila6x4943hiyyizc55kp85aw0imn";
+    url = "https://hub-dist.unity3d.com/artifactory/hub-debian-prod-local/pool/main/u/unity/unityhub_amd64/unityhub-amd64-${version}.deb";
+    sha256 = "sha256-/P6gPLSRGfwEN801cyNrZTpHyZKO+4tU6cFvLz8ERuo=";
   };
 
-  extraPkgs = (pkgs: with pkgs; with xorg; [ gtk2 gdk-pixbuf glib libGL libGLU nss nspr
-    alsa-lib cups libcap fontconfig freetype pango
-    cairo dbus dbus-glib libdbusmenu libdbusmenu-gtk2 expat zlib libpng12 udev tbb
-    libpqxx gtk3 libsecret lsb-release openssl nodejs ncurses5
+  nativeBuildInputs = [
+    dpkg
+    makeWrapper
+  ];
 
-    libX11 libXcursor libXdamage libXfixes libXrender libXi
-    libXcomposite libXext libXrandr libXtst libSM libICE libxcb
+  fhsEnv = buildFHSUserEnv {
+    name = "${name}-fhs-env";
+    runScript = "";
+    targetPkgs = pkgs: with pkgs; [
+      # Unity Hub dependencies
+      cups
+      gtk3
+      expat
+      libxkbcommon
+      lttng-ust_2_12
+      krb5
+      alsa-lib
+      nss_latest
+      libdrm
+      mesa
+      nspr
+      atk
+      dbus
+      at-spi2-core
+      # at-spi2-atk - at-spi2-core used to be split up into 2 packages
+      pango
+      xorg.libXcomposite
+      xorg.libXrandr
+      xorg.libXext
+      xorg.libXdamage
+      xorg.libXfixes
+      xorg.libxcb
+      xorg.libxshmfence
+      xorg.libXScrnSaver
+      xorg.libXtst
 
-    libselinux pciutils libpulseaudio libxml2 icu clang cacert
-  ]);
+      libva
+      openssl_1_1
+      cairo
+      xdg-utils
+      libnotify
+      libuuid
+      libsecret
+      udev
+      libappindicator
+      wayland
+      cpio
+      icu
+      libpulseaudio # Not a ldd thing, but needed for sound to work
 
-  extraInstallCommands =
-    let appimageContents = appimageTools.extractType2 { inherit pname version src; }; in
-    ''
-      install -Dm444 ${appimageContents}/unityhub.desktop -t $out/share/applications
-      substituteInPlace $out/share/applications/unityhub.desktop \
-        --replace 'Exec=AppRun' 'Exec=${pname}'
-      install -m 444 -D ${appimageContents}/unityhub.png \
-        $out/share/icons/hicolor/64x64/apps/unityhub.png
-    '';
+      # Editor dependencies
+      libglvnd # provides ligbl
+      xorg.libX11
+      xorg.libXcursor
+      glib
+      gdk-pixbuf
+      libxml2
+      zlib
+      clang
+
+      # Bug Reporter dependencies
+      fontconfig
+      freetype
+      lsb-release
+    ] ++ extraPkgs pkgs;
+  };
+
+  unpackCmd = ''
+    runHook preUnpack
+
+    dpkg-deb -x $src src
+
+    runHook postUnpack
+  '';
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out
+    mv opt/ usr/share/ $out
+
+    # `unityhub` is a shell wrapper that runs `unityhub-bin`
+    # Which we don't need and replace with our own custom wrapper
```

We should probably remove the unused wrapper

---

`makeWrapper` should overwrite the existing wrapper, but it still might be worth removing it outright first for clarity.
```suggestion
    # `unityhub` is a shell wrapper that runs `unityhub-bin`
    # Which we don't need and replace with our own custom wrapper
    rm$out/opt/unityhub/unityhub
```